packages/utils/src/internal-utils.ts

```diff
@@ -50,22 +37,44 @@ export const collector = <K, V, VV>(
     | { t: CollectorAction.Delete; k: K }
     | { t: CollectorAction.Set; k: K; v: VV }
   >,
-): ((source: Observable<GroupedObservable<K, V>>) => Observable<Map<K, VV>>) =>
-  pipe(
-    publish((x) => x.pipe(mergeMap(enhancer), takeUntil(takeLast(1)(x)))),
-    endWith({ t: CollectorAction.Complete as const }),
-    scanWithDefaultValue(
-      (acc, val) => {
-        if (val.t === CollectorAction.Set) {
-          acc.set(val.k, val.v)
-        } else if (val.t === CollectorAction.Delete) {
-          acc.delete(val.k)
-        } else {
-          acc.clear()
-        }
-        return acc
-      },
-      () => new Map<K, VV>(),
-    ),
-    shareLatest(),
-  )
+): ((
+  source: Observable<GroupedObservable<K, V>>,
+) => Observable<Map<K, VV>>) => (source$) =>
+  new Observable<Map<K, VV>>((observer) => {
+    const subscription = new Subscription()
+    const map = new Map<K, VV>()
+    let emitted = false
+
+    subscription.add(
+      source$.subscribe(
+        (x) => {
+          subscription.add(
+            enhancer(x).subscribe(
+              (v) => {
+                if (v.t === CollectorAction.Set) {
+                  map.set(v.k, v.v)
+                } else {
+                  map.delete(v.k)
+                }
+                emitted = true
+                observer.next(map)
+              },
+              (e) => {
+                observer.error(e)
+              },
+            ),
```

Comment:
I'm thinking that both collect and collectValues delete the key only on complete or error (except that in collectValues it propagates the error too) - Maybe this enhancer thing could be further simplified if it only emits `CollectorAction.Set` events instead?

e.g.
```ts
enhancer(x).subscribe(
   ([k,v]) => {
      map.set(k,v);
      emitted = true
      observer.next(map)
   },
   e => {
     observer.error(e)
   },
   () => {
      map.delete( /*.... errr where do I get the key from? 😅 */)
   }
``` 

Mhh maybe not 😅 - Or yes, isn't the `x ` a GroupObservable? That would have the key.

Reply:
That's a very good point... Now that we got rid of the `filter` of `collect`, this can be simplified quite a lot. Let me think about it, thanks for pointing this out!

Reply:
Check the latest commit! Thanks a lot for this comment!